nextflow/src/org/labkey/nextflow/NextFlowController.java

```diff
@@ -0,0 +1,307 @@
+                {
+                    map.put(IS_NEXTFLOW_ENABLED, Boolean.FALSE.toString());
+                }
+                else
+                {
+                    map.put(IS_NEXTFLOW_ENABLED, Boolean.TRUE.toString());
+                }
+            }
+            map.save();
+            return true;
+        }
+
+        @Override
+        public void addNavTrail(NavTree root)
+        {
+
+        }
+
+        @Override
+        public URLHelper getSuccessURL(Object o)
+        {
+            return getContainer().getStartURL(getUser());
+        }
+    }
+
+    @RequiresPermission(AdminOperationsPermission.class)
+    public class NextFlowRunAction extends FormViewAction
+    {
+        private ActionURL _successURL;
+        @Override
+        public void validateCommand(Object o, Errors errors)
+        {
+            PropertyStore store = PropertyManager.getNormalStore();
+            PropertyManager.PropertyMap map = store.getWritableProperties(NextFlowManager.NEXTFLOW_ENABLE, false);
+            if (map == null || !"true".equals(map.get("enabled")))
+            {
+                errors.reject(ERROR_MSG, "NextFlow is not enabled");
+            }
+        }
+
+        @Override
+        public ModelAndView getView(Object o, boolean b, BindException errors) throws Exception
+        {
+            return new HtmlView("NextFlow Runner", DIV("Run NextFlow Pipeline",
+                    FORM(at(method, "POST"),
+                            new Button.ButtonBuilder("Start NextFlow").submit(true).build())));
+        }
+
+        @Override
+        public boolean handlePost(Object o, BindException errors) throws Exception
+        {
+            // check if nextflow is enabled
+            PropertyStore store = PropertyManager.getNormalStore();
+            PropertyManager.PropertyMap map = store.getProperties(NextFlowManager.NEXTFLOW_ENABLE);
+            if (map == null || !Boolean.parseBoolean(map.get(IS_NEXTFLOW_ENABLED)))
+            {
+                errors.reject(ERROR_MSG, "NextFlow is not enabled");
+                return false;
+            }
+
+            try (SecurityManager.TransformSession session = SecurityManager.createTransformSession(getViewContext()))
+            {
+                // TODO: pass the apiKey to Nextflow job
+                String apiKey = session.getApiKey();
+                ViewBackgroundInfo info = getViewBackgroundInfo();
+                PipeRoot root = PipelineService.get().findPipelineRoot(info.getContainer());
+                PipelineJob job = new NextFlowPipelineJob(info, root, apiKey);
+                PipelineService.get().queueJob(job);
+            }
+
+            return !errors.hasErrors();
+        }
+
+        @Override
+        public URLHelper getSuccessURL(Object o)
+        {
+            return PageFlowUtil.urlProvider(PipelineStatusUrls.class).urlBegin(getContainer());
+        }
+
+        @Override
+        public void addNavTrail(NavTree navTree)
+        {
+
```

Comment:
Append to the nav trail to give the page a title

Reply:
done